docs/scaling.md

```diff
@@ -0,0 +1,265 @@
+For this you can use the `ThreadPoolExecutor` or `ProcessPoolExecutor` class from the `concurrent.futures` module in the python standard library.
+You simply pass the executor class directly via the `parallel` kwarg.
+
+```python
+from concurrent.futures import ThreadPoolExecutor
+
+combined_vds = vz.open_virtual_mfdataset(filepaths, parallel=ThreadPoolExecutor)
+```
+
+This can work well when virtualizing files in remote object storage because it parallelizes the issuing of HTTP GET requests for each file.
+
+### Dask Delayed
+
+You can parallelize using `dask.delayed` automatically by passing `parallel='dask'`.
+This will select the `virtualizarr.parallel.DaskDelayedExecutor`.
+
+```python
+combined_vds = vz.open_virtual_mfdataset(filepaths, parallel='dask')
+```
+
+This uses the same approach that `xarray.open_mfdataset` does when `parallel=True` is passed to it.
+Using `dask.delayed` allows for parallelizing with any type of dask cluster, included a managed [Coiled](http://www.coiled.io) cluster.
+
+### Lithops
+
+As the map step is totally embarrassingly parallel, it can be performed entirely using serverless functions.
+This approach allows for virtualizing N files in the same time it takes to virtualize 1 file, (assuming you can provision N concurrent serverless functions), avoiding the need to configure, scale, and shutdown a cluster.
+
+You can parallelize VirtualiZarr serverlessly by using the [lithops](http://lithops-cloud.github.io) library.
+Lithops can run on all the main cloud provider's serverless FaaS platforms.
+
+To run on lithops you need to configure lithops for the relevant compute backend (e.g. AWS Lambda), build a runtime using Docker ([example Dockerfile]() with the required dependencies), and ensure the necessary cloud permissions to run are available.
+Then you can use the `virtualizarr.parallel.LithopsEagerFunctionExecutor` simply via
+
+```python
+combined_vds = vz.open_virtual_mfdataset(filepaths, parallel='lithops')
+```
+
+### Custom Executors
+
+You can also define your own executor to run in some other way, for example on a different serverless platform such as [Modal](https://modal.com).
+
+Your custom executor must inherit from the `concurrent.futures.Executor` ABC, and must implement the `.map` method.
+
+```python
+from concurrent.futures import Executor
+
+class CustomExecutor(Executor):
+    def map(
+        self,
+        fn: Callable,
+        *iterables: Iterable,
+    ) -> Iterator:
+        ...
+
+combined_vds = vz.open_virtual_mfdataset(filepaths, parallel=CustomExecutor)
+```
+
+## Memory usage
+
+For the virtualization to succeed you need to ensure that your available memory is not exceeded at any point.
+There are 3 points at which this might happen:
+
+1. While generating references
+2. While combining references
+3. While writing references
+
+While generating references each worker calling `open_virtual_dataset` needs to avoid running out of memory.
```

Comment:
```suggestion
While generating references each worker calling [`open_virtual_dataset`][virtualizarr.open_virtual_dataset] needs to avoid running out of memory.
```